src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -355,9 +335,93 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 		// eat up rest of line
 		bufPtr = goNextLine(bufPtr, bufEnd);
 	}	// end while(bufPtr && (bufPtr-buf<filesize))
-	// Clean up the allocate obj _file contents
-	delete [] buf;
 
+    core::vector<core::vectorSIMDf> newNormals;
+    auto doRecalcNormals = [&vertices,&newNormals](const core::vector<uint32_t>& _ixs) {
+        memset(newNormals.data(), 0, sizeof(core::vectorSIMDf)*newNormals.size());
+
+        auto minmax = std::minmax_element(_ixs.begin(), _ixs.end());
+        const uint32_t maxsz = *minmax.second - *minmax.first;
+        const uint32_t min = *minmax.first;
+        
+        newNormals.resize(maxsz, core::vectorSIMDf(0.f));
+        for (size_t i = 0ull; i < _ixs.size(); i += 3ull)
+        {
+            core::vectorSIMDf v1, v2, v3;
+            v1.set(vertices[_ixs[i-min+0u]].pos);
+            v2.set(vertices[_ixs[i-min+1u]].pos);
+            v3.set(vertices[_ixs[i-min+2u]].pos);
+            v1.makeSafe3D();
+            v2.makeSafe3D();
+            v3.makeSafe3D();
+            core::vectorSIMDf normal(core::plane3dSIMDf(v1, v2, v3).getNormal());
+            newNormals[_ixs[i-min+0u]] += normal;
+            newNormals[_ixs[i-min+1u]] += normal;
+            newNormals[_ixs[i-min+2u]] += normal;
+        }
+        for (uint32_t ix : _ixs)
+            vertices[ix].normal32bit = asset::quantizeNormal2_10_10_10(newNormals[ix-min]);
+    };
```

Comment:
we should probably use the @Przemog1 utility function for this XD

Reply:
what exactly are you talking about?

Reply:
hmm.. I think I got confused, although I believe I could have been thinking about the need of the normals to be flipped for RH meshes (different winding order for front-face so different cross product sign needed)